sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_internal/rsa_key.py

```diff
@@ -214,7 +213,10 @@ def unwrap_key(self, encrypted_key, **kwargs):
         return decryptor.transform(encrypted_key)
 
     def is_private_key(self):
-        return isinstance(self._rsa_impl, RSAPrivateKey)
```

Comment:
As we briefly discussed OOB, this makes me leery, even though I understand why we'd need to... I wonder if there's any way to expose this risk/let the user know that it's a lie, without spamming end users all over the place.  (My only fear is that in the future some poor internal dev will be trying to figure out "WHY IS THE LIB LYING" and wouldn't know unless they actually asked the question-we're-all-trained-never-to-ask-in-favor-of-blaming-ourselves of "is my tool broken")

Reply:
Does it help that none of the `_internal` code is part of the public API? We never return these types to users.

Reply:
Oh sure, it moots the problem from a user-facing perspective, at this point I'm thinking exclusively "long term code ownership/maintenance"-wise.  I think even a docstring that a dev would see when trying to consume this to warn them that it's smoke-and-mirrors would quell my fears.